core/comp_ui.py

```diff
@@ -142,41 +146,46 @@ class MinimalDisplay(_IDisplay):
     without testing it.
     """
 
-    def __init__(self, comp_state, prompt_state, debug_f):
-        # type: (State, PromptState, _DebugFile) -> None
+    def __init__(self, comp_state, prompt_state, debug_f, term_width, signal_safe):
+        # type: (State, PromptState, _DebugFile, int, iolib.SignalSafe) -> None
         _IDisplay.__init__(self, comp_state, prompt_state, 10, mylib.Stdout(),
                            debug_f)
 
+        self.signal_safe = signal_safe
+        self.term_width = term_width
+
     def _RedrawPrompt(self):
         # type: () -> None
         # NOTE: This has to reprint the prompt and the command line!
         # Like bash, we SAVE the prompt and print it, rather than re-evaluating it.
         self.f.write(self.prompt_state.last_prompt_str)
         self.f.write(self.comp_state.line_until_tab)
 
+    def _GetTerminalWidth(self):
```

Comment:
I think we can move this _GetTerminalWidth method up to the base class _IDisplay?